src/NSwag.AspNetCore/IDocumentProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IDocumentProvider.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ApiDescription
+{
+    // This service will be looked up by name from the service collection when using
+    // the Microsoft.Extensions.ApiDescription tool
+    internal interface IDocumentProvider
```

Comment:
I think we should rename it to IOpenApiDocumentProvider because it's specific to Swagger/OpenAPI3 documents. Otherwise I'd introduce a type enum just to be future proof.

Reply:
What do you think?

Reply:
Though I could be convinced, I'm fairly resistant to this idea.

Nothing in the `Microsoft.Extensions.ApiDescription.Design` package is specific to Open API or the earlier Swagger formats. All that's required is that the service return a document the code generator can handle. And, the various documents are all self-describing.

If NSwag supports new formats, I suggest using distinct `DocumentName`s for the different formats.

Reply:
Tool can now find this `internal` type. Turned out the issue wasn't the visibility modifier at all.